frontend/src/lib/components/ActivityLog/SentenceList.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+
+interface SentenceListProps {
+    listParts: (string | JSX.Element | null)[]
+    prefix?: string | JSX.Element | null
+    suffix?: string | JSX.Element | null
+}
+
+export function SentenceList({ listParts, prefix = null, suffix = null }: SentenceListProps): JSX.Element {
```

Comment:
extracting this was well overdue 🙌  Definitely needs at least a story book page though